src/availability.rs

```diff
@@ -472,10 +475,117 @@ where
                 Ok(result)
             }
             .boxed()
+        })?
+        .get("get_transaction_range_rev", move |req, state| {
+            async move {
+                let height: usize = req.integer_param("height")?;
+                let offset: usize = req.integer_param("offset")?;
+                let limit: usize = req.integer_param("limit")?;
+                let batch_size = NonZeroUsize::new(10).unwrap();
+
+                let chunked_ranges = build_range_iterator_rev(height, batch_size);
+                let stream_of_chunked_blocks = futures::stream::iter(chunked_ranges)
+                    .then(move |range| async move {
+                        let stream = expand_range_to_block_range_stream::<State, Types>(
+                            range, state, timeout,
+                        )
+                        .await;
+
+                        let vec_of_blocks = stream
+                            .collect::<Vec<Result<BlockQueryData<Types>, Error>>>()
+                            .await;
+
+                        // We need to reverse here, otherwise we're not in the correct order.
+                        vec_of_blocks.into_iter().rev()
+                    })
+                    .flat_map(futures::stream::iter);
+
+                let chunked_transactions = stream_of_chunked_blocks.map(|result| match result {
+                    Ok(block) => {
+                        let transactions = expand_block_data_to_transaction_vec(block);
+                        transactions
+                            .into_iter()
+                            .rev()
+                            .map(Ok)
+                            .collect::<Vec<Result<Transaction<Types>, Error>>>()
+                    }
+                    Err(err) => vec![Err(err)],
+                });
+
+                let streamed_transactions =
+                    chunked_transactions.flat_map(|vec| futures::stream::iter(vec.into_iter()));
+
+                let result = streamed_transactions
+                    .skip(offset)
+                    .take(limit)
+                    .try_collect::<Vec<Transaction<Types>>>()
+                    .await?;
+
+                Ok(result)
+            }
+            .boxed()
         })?;
     Ok(api)
 }
 
+// build_range_iterator_rev is a helper function that builds iterator of ranges
```

Comment:
Not sure if you meant for this, but using `//` instead of `///` results in no documentation showing up when applying `hover` within an editor. To fix this, you'll just need to change this comment (and the other functions below) to use `///` and it'll give hover support for the function.

Reply:
No worries @jparr721! I'm new to `rust` so I appreciate the insight!